### PR TITLE
Add FlashComponent Session getter.

### PR DESCRIPTION
**Pull request by robertpustulka**
Refs #12958

This prevents a situation when the outdated session object is referenced by a controller.
Since PSR-7 request and response objects are immutable, request may be changed at runtime along with the session object.

### CODE REVIEW

File: src/Controller/Component/FlashComponent.php [hunk 1]
```diff
@@ -31,13 +30,6 @@
 class FlashComponent extends Component
 {
 
-    /**
-     * The Session object instance
-     *
-     * @var \Cake\Http\Session
-     */
-    protected $_session;
```
Comment (saeideng) [inline src/Controller/Component/FlashComponent.php]: you can not delete `protected` member in bugfix,  for BC reasons

Comment (dereuromark) [reply src/Controller/Component/FlashComponent.php]: We could have __...() to proxy then, right?
But yeah, people could be using the property internally right now.

File: src/Controller/Component/FlashComponent.php [hunk 2]
```diff
@@ -51,18 +43,6 @@ class FlashComponent extends Component
...
-    /**
-     * Constructor
-     *
-     * @param \Cake\Controller\ComponentRegistry $registry A ComponentRegistry for this component
-     * @param array $config Array of config.
-     */
-    public function __construct(ComponentRegistry $registry, array $config = [])
-    {
-        parent::__construct($registry, $config);
-        $this->_session = $registry->getController()->getRequest()->getSession();
```
Comment (markstory) [inline src/Controller/Component/FlashComponent.php]: We should leave this behind and just stop using it internally. App extensions might still be using it.

File: src/Controller/Component/FlashComponent.php (hunk 1 above)
Comment (saeideng) [reply src/Controller/Component/FlashComponent.php]: now we can add a note for this that it will remove in 4.x

**Comment by dereuromark** [comment]
You could still add a helping deprecation to

    protected $_session;

though

**Comment by robertpustulka** [comment]
This should probably be targeting 3.8.0

**Comment by markstory** [comment]
> This should probably be targeting 3.8.0

Why? It looks like a pretty safe change to me.

**Review by dereuromark — APPROVED** [review]

**Pull request merged**